test/suites/standard/systemd-resolved.robot

```diff
@@ -0,0 +1,120 @@
+*** Settings ***
+Documentation       Verify MicroShift host name resolution with systemd-resolved installed
+
+Resource            ../../resources/common.resource
+Resource            ../../resources/microshift-host.resource
+Resource            ../../resources/microshift-process.resource
+Resource            ../../resources/ostree-health.resource
+Resource            ../../resources/systemd.resource
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+
+*** Variables ***
+${KUBELET_CONFIG_FILE}      /var/lib/microshift/resources/kubelet/config/config.yaml
+
+
+*** Test Cases ***
+Verify Kubelet Config With and Wihtout Systemd-Resolved
```

Comment:
Can we split this into 2 tests so we can tell from the failure which mode is not working?

Reply:
Currently, the test needs to be run in this order because we start with a situation with `systemd-resolved` installed. 
Can you recommend on how to enforce the order if we split the tests?

Reply:
We usually writes tests to enforce their own preconditions with setup steps. So in the test where we want to ensure it works when the package is installed, have a setup to install it that does not fail if it is already installed. In the test where we want to ensure it works when the package is not installed, do the opposite. Then the tests can call a common keyword to ensure that DNS resolution works as expected.

Reply:
The problem is that we cannot really install this package unless we create a special commit for it for ostree configuration. I did not want to further slow down the image build procedure because of this simple test.

Reply:
Why can we uninstall the package, but not install it?

Reply:
- We can easily install/uninstall on a non-ostree system using `dnf`.
- On an ostree system:
  - we can easily uninstall using `rpm-ostree override`.
  - we can also install using the same `rpm-ostree override`, but we need the RPM file, which we cannot get easily using "standard" repos

So, we should either include the package in a commit, or copy RPM to the system without installing it.
Is there anything I'm missing?

Reply:
I was able to split the test sequence into 3 distinct tests now that we understand what we should expect.